relay-server/src/services/store.rs

```diff
@@ -1275,12 +1313,58 @@ struct MetricKafkaMessage<'a> {
     project_id: ProjectId,
     name: &'a str,
     #[serde(flatten)]
-    value: BucketViewValue<'a>,
+    value: MetricValue<'a>,
     timestamp: UnixTimestamp,
     tags: &'a BTreeMap<String, String>,
     retention_days: u16,
 }
 
+#[derive(Clone, Debug, Serialize)]
+#[serde(tag = "type", content = "value")]
+enum MetricValue<'a> {
+    #[serde(rename = "c")]
+    Counter(FiniteF64),
+    #[serde(rename = "d")]
+    Distribution(FloatArrayEncoding<&'a [FiniteF64]>),
+    #[serde(rename = "s")]
+    Set(FloatArrayEncoding<SetView<'a>>),
```

Comment:
nit(naming): `Set` does not contain floats, so should we rename to `NumericArrayEncoding`?

Reply:
Renamed to just `ArrayEncoding`, since that is actually what it is, the code does not care about the payload 👍 .